File: python/3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        right = 0
        
        ans = 0
        
        letter = [0]*256
        while right < len(s):
            c = s[right]
            letter[ord(c)]+=1
            
            while letter[ord(c)] > 1:
                t = s[left]
                letter[ord(t)] -=1
                left+=1
            
            ans = max(ans, right-left+1)
            right+=1
        return ans
            
            

File: golang/3.go
func max(a,b int) int {
    if a<b {
        return b
    }
    return a
}
func lengthOfLongestSubstring(s string) int {
    letter := make([]int, 256, 256)
    left := 0
    right:=0
    ans := 0
    
    arChar := []rune(s)
    for ;right<len(arChar); {
        letter[arChar[right]]++
        for ;letter[arChar[right]]>1;{
            letter[arChar[left]]--
            left++
        }
        ans = max(ans, right-left+1)
        right++
    }
    return ans
}
